site/src/pages/[shortname]/[blockSlug].page.tsx

```diff
@@ -114,35 +112,20 @@ export const getStaticProps: GetStaticProps<
     return { notFound: true };
   }
 
+  const packagePath = `${shortname}/${blockSlug}`;
   const catalog = readBlocksFromDisk();
 
-  const blockMetadataResponse = await fetch(
-    `${BASE_URL}/blocks/${shortname}/${blockSlug}/block-metadata.json`,
+  const blockMetadata = catalog.find(
+    (metadata) => metadata.packagePath === packagePath,
   );
 
-  if (blockMetadataResponse.status === 404) {
+  if (!blockMetadata) {
     // TODO: Render custom 404 page for blocks
     return { notFound: true };
-  } else if (!blockMetadataResponse.ok) {
-    // TODO: Render details of upstream error
-    throw new Error("Something went wrong");
   }
 
-  // TODO: handle malformed block metadata
-  const blockMetadata: BlockMetadata = await blockMetadataResponse.json();
-
-  blockMetadata.lastUpdated = catalog.find(
-    ({ name }) => name === blockMetadata.name,
-  )?.lastUpdated;
-
-  const [schema, blockStringifiedSource] = await Promise.all([
-    fetch(
-      `${BASE_URL}/blocks/${shortname}/${blockSlug}/${blockMetadata.schema}`,
-    ).then((res) => res.json()),
-    fetch(
-      `${BASE_URL}/blocks/${shortname}/${blockSlug}/${blockMetadata.source}`,
-    ).then((res) => res.text()),
-  ]);
```

Comment:
This changes are already in main. Just brought them over since I didn't want to pull all the changes there